test/controllers/sections/items/ItemDegreesPlatoControllerSpec.scala

```diff
@@ -0,0 +1,163 @@
+      }
+    }
+
+    "must redirect to Index of section when the idx is outside of bounds for a POST" in new Fixture() {
+      running(application) {
+
+        val request = FakeRequest(POST, itemDegreesPlatoRoute(testIndex2)).withFormUrlEncodedBody((hasDegreesPlatoField, "false"))
+        val result = route(application, request).value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustBe routes.ItemsIndexController.onPageLoad(testErn, testDraftId).url
+      }
+    }
+
+    "must return OK and the correct view for a GET" in new Fixture() {
+      running(application) {
+
+        val request = FakeRequest(GET, itemDegreesPlatoRoute())
+        val result = route(application, request).value
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(form, itemDegreesPlatoSubmitAction(), Some(Wine))(dataRequest(request), messages(application)).toString
+      }
+    }
+
+    "must populate the view correctly on a GET when the question has previously been answered" in new Fixture(
+      Some(defaultUserAnswers.set(ItemDegreesPlatoPage(testIndex1), ItemDegreesPlatoModel(hasDegreesPlato = true, Some(5))))
+    ) {
+      running(application) {
+
+        val request = FakeRequest(GET, itemDegreesPlatoRoute())
+        val result = route(application, request).value
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(
+          form.fill(ItemDegreesPlatoModel(hasDegreesPlato = true, Some(5))),
+          itemDegreesPlatoSubmitAction(),
+          Some(Wine)
+        )(dataRequest(request), messages(application)).toString
+      }
+    }
+
+    "must redirect to the next page when valid data is submitted" in new Fixture() {
+      running(application) {
+
+        MockUserAnswersService.set().returns(Future.successful(emptyUserAnswers))
+
+        val request = FakeRequest(POST, itemDegreesPlatoRoute()).withFormUrlEncodedBody((hasDegreesPlatoField, "false"))
+        val result = route(application, request).value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual testOnwardRoute.url
+      }
+    }
+
+    "must return a Bad Request and errors when invalid data is submitted (no brand name supplied)" in new Fixture() {
+      running(application) {
+
+        val request = FakeRequest(POST, itemDegreesPlatoRoute()).withFormUrlEncodedBody((hasDegreesPlatoField, "true"))
+        val boundForm =
+          form
+            .bind(Map(hasDegreesPlatoField -> "true"))
+            .withError(degreesPlatoField, ItemDegreesPlatoMessages.English.errorDegreesPlatoRequired)
+        val result = route(application, request).value
+
+        status(result) mustEqual BAD_REQUEST
+        contentAsString(result) mustEqual view(boundForm, itemDegreesPlatoSubmitAction(), Some(Wine))(dataRequest(request), messages(application)).toString
+      }
+    }
```

Comment:
I think you need another BadRequest test for the normal error too, at the moment you only check the BadRequest you create as part of the `handleSubmittedForm`